src/Xamarin.Android.Build.Tasks/Utilities/MamJsonParser.cs

```diff
@@ -0,0 +1,215 @@
+>;
+
+namespace Xamarin.Android.Tasks
+{
+	class MamJsonParser
+	{
+		Action<TraceLevel, string>  Logger;
+
+		public	readonly    ReplacementTypesDict    ReplacementTypes        = new ();
+		public  readonly    ReplacementMethodsDict  ReplacementMethods      = new ();
+
+		public MamJsonParser (Action<TraceLevel, string> logger)
+		{
+			Logger  = logger;
+		}
+
+		public void Load (string jsonPath)
+		{
+			var json    = ReadJson (jsonPath);
+
+			var classRewrites = json["ClassRewrites"];
+			if (classRewrites != null) {
+				ReadClassRewrites (classRewrites);
+			}
+
+			var globalMethodCalls = json["GlobalMethodCalls"];
+			if (globalMethodCalls != null) {
+				ReadGlobalMethodCalls (globalMethodCalls);
+			}
+		}
+
+		public XElement ToXml ()
+		{
+			return new XElement ("replacements",
+				GetReplacementTypes (),
+				GetReplacementMethods ());
+		}
+
+		static JObject ReadJson (string path)
+		{
+			using (var f = File.OpenText (path))
+			using (var r = new JsonTextReader (f))
+				return (JObject) JToken.ReadFrom (r);
+		}
+
+		void ReadClassRewrites (JToken classRewrites)
+		{
+			foreach (var classRewrite in classRewrites) {
+				if (!TryReadClassFromTo (classRewrite, out var from, out var to)) {
+					Logger (TraceLevel.Verbose, $"No from or to! {classRewrite}");
+					continue;
+				}
+				ReplacementTypes [from] = to;
+				var methods = classRewrite ["Methods"];
+				if (methods == null) {
+					continue;
+				}
+				foreach (var method in methods) {
+					var makeStatic = (bool?) method["MakeStatic"] ?? false;
+					var oldName = (string?) method["OriginalName"];
+					var newName = (string?) method["NewName"];
+					var oldSig  = ReadSignature (method["OriginalParams"]);
+					if (oldName == null || newName == null) {
+						continue;
+					}
+					ReplacementMethods [(to, oldName, oldSig)] = (to, newName, null, null, makeStatic);
+				}
+			}
+		}
+
+		bool TryReadClassFromTo (JToken token, [NotNullWhen(true)] out string? from, [NotNullWhen(true)] out string? to)
+		{
+			from    = (string?) token["Class"]?["From"];
+			to      = (string?) token["Class"]?["To"];
+			if (from == null || to == null) {
+				return false;
+			}
+			from = JavaToJniType (from);
+			to = JavaToJniType (to);
+			return true;
+		}
+
+		string? ReadSignature (JToken? token)
+		{
+			if (token == null)
+				return null;
+			var types = new List<string> ();
+			foreach (var type in token) {
+				if (type == null) {
+					continue;
+				}
+				var javaType = ((string?) type) switch {
+					"boolean"   => "Z",
+					"byte"      => "B",
+					"char"      => "C",
+					"double"    => "D",
+					"float"     => "F",
+					"int"       => "I",
+					"long"      => "J",
+					"short"     => "S",
+					"void"      => "V",
+					var o       => JavaToJniTypeSignature (o),
+				};
+				if (javaType == null) {
+					continue;
+				}
+				types.Add (javaType);
+			}
+			if (types.Count == 0)
+				return null;
+			return "(" + string.Join ("", types) + ")";
```

Comment:
I wonder if using a `StringBuilder` here (and getting rid fo the `types` list) would be a bit faster?

Reply:
Updated to use `StringBuilder`.